### ISSUE TITLE
Multiple fine-tuning jobs with chain-job restart at `istep=0`

### ISSUE THREAD

**Issue by MatKbauer** [ecmwf/WeatherGenerator#1448]
### What happened?

When fine-tuning a pre-trained model, I use the following command on santis
```
../WeatherGenerator-private/hpc/launch-slurm.py --nodes 2 --from-run-id d583uri6 --chain-jobs 2 --options istep=0 num_epochs=32 forecast_steps=8
```
This works well for the first job, but the second job in the chain restarts the training at epoch 0 instead of convening the training where the previous job finished. Probably because the `istep=0` is passed to all chained jobs.

To resolve this, we can pass `istep=0` only to the first job in the chain. In fact, it should work to pass all `--options` to the very first job only, because subsequent jobs will pick up those options from the config of the first job.

@simone99n, maybe you have some ideas how to approach this?

### What are the steps to reproduce the bug?

Checkout `/mk/develop/fe_experiments` on `santis` and run the command above. Maybe add `--options samples_per_epoch=64 samples_per_validation=32 num_epochs=4` to the job to speed up the troubleshooting.

**Comment by simone99n** [comment]
assign me

**Comment by simone99n** [comment]
@MatKbauer Could you test my quick fix in `simone99n/dev/1448_option_args` (private repo)?
It should work. If not, I'll investigate further

**Comment by grassesi** [comment]
@simone99n Can you please make a PR in the private repo and link it here?

**Comment by simone99n** [comment]
Done. See merge request [!103](https://gitlab.jsc.fz-juelich.de/esde/WeatherGenerator-private/-/merge_requests/103/diffs)

**Comment by sophie-xhonneux** [comment]
I still get this error as you can see by this learning rate problem. Happens for train-continue with chaining jobs

<img width="830" height="587" alt="Image" src="https://github.com/user-attachments/assets/d9e71408-902d-488f-ab84-5808aca1f6f4" />

**Comment by simone99n** [comment]
> I still get this error as you can see by this learning rate problem. Happens for train-continue with chaining jobs

Oh, I see. I've done a quick debug and I prepared [PR !152](https://gitlab.jsc.fz-juelich.de/esde/WeatherGenerator-private/-/merge_requests/152).
@sophie-xhonneux Could you test it? 

@grassesi  should remove the use of the environment variable `WEATHERGEN_STAGE` in the future? I see your TODO comment [here](https://github.com/ecmwf/WeatherGenerator/blob/develop/src/weathergen/run_train.py#L64)

**Comment by sophie-xhonneux** [comment]
@simone99n Are you sure that is the error because I checked the sbatch commands that caused this issue (see attached screenshots)? and it looks to me like the training stage was correctly set

<img width="1512" height="180" alt="Image" src="https://github.com/user-attachments/assets/a84e51d5-0a85-4aa1-b496-2d0550a85861" />

**Comment by simone99n** [comment]
Seems trickier than expected...
Tomorrow I'll have a look.

**Comment by simone99n** [comment]
> [@simone99n](https://github.com/simone99n) Are you sure that is the error because I checked the sbatch commands that caused this issue (see attached screenshots)? and it looks to me like the training stage was correctly set
> 
> <img alt="Image" width="1512" height="180" src="https://private-user-images.githubusercontent.com/24638638/574752219-a84e51d5-0a85-4aa1-b496-2d0550a85861.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzU2MzQ5MjcsIm5iZiI6MTc3NTYzNDYyNywicGF0aCI6Ii8yNDYzODYzOC81NzQ3NTIyMTktYTg0ZTUxZDUtMGE4NS00YWExLWI0OTYtMmQwNTUwYTg1ODYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDA4VDA3NTAyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjYzNiZjU3ZWYyMmI4MDNiZjQ4ZmRhMDMwMDlmOGE0YTFhZDMxNjEyZjIwZjViZjI3YmZlN2JkYjdiYjlhMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xDNZiH_LSMYFghKgNklVHD2Tk8M9xyh1RI-0QZsJxiY">

A couple of questions:
- Are you using the latest version of the private repo?
It's very strange to me for see `WEATHERGENERATOR_STAGE=train_continue` in the logs... yesterday, I found a bug in launch-slurm that always set `WEATHERGENERATOR_STAGE=train` (fixed with [MR!152](https://gitlab.jsc.fz-juelich.de/esde/WeatherGenerator-private/-/merge_requests/152/diffs))
- Can you provide me the launch command / the pipeline config file?
I want to reproduce the bug

**Comment by grassesi** [comment]
@simone99n @sophie-xhonneux What is the status of this, Can this be closed in favor of #2205 ?

**Comment by MatKbauer** [comment]
I tested with the latest develop and looks good to me. Both the manually submitted train_continue slurm job and the pipelined jobs exhibited the desired behavior without resetting `istep` during job-continuation

<img width="300" height="210" alt="Image" src="https://github.com/user-attachments/assets/ebeb186c-f690-42a2-8c39-84dfeceb86f8" />
<img width="300" height="210" alt="Image" src="https://github.com/user-attachments/assets/b8afcf51-a3ff-455a-aab8-a20b0cc3fa5f" />

Closing this issue therefore again, but #2205 should still be verified.
